backend/framework/qlf/dashboard/bokeh/qlf_plot.py

```diff
@@ -204,6 +204,171 @@ def metric_table(metricname, comments, keyname,  curexp='xxx', refexp='xxx', obj
+        </tr>"""
+
+    end="""</table> </div> """
+
+    title="""<h2 align=center style="text-align:center;">  {} </h2>""".format(qa)
+
+
+    if qa=='countpix':
+        ref=['N/A']*4
+        nrows=4
+    else:
+        ref=par[key+'_REF']
+        if isinstance(ref,list):
+            nrows=len(ref)
+        else:
+            nrows=1
+    
+    try:
+        curexp=met[key]
+    except Exception as err:
+        print('ERR {}'.format(err))
+        
+        tblines="""<tr>
+                <td colspan="4">Error in key:{}</td>
+                </tr>
+                """.format(err)
+
+        return style + title + header +tblines+ end
+    
+    if nrows > 1:
+        cur_tb, ref_tb=[],[]
+        for i in list(range(nrows)):
+            x = curexp[i]
+            ref_tb.append(ref[i])
+            
+            if isinstance(x, float) and x > -999:
+                xstr='%4.3f'%(x)
+            elif isinstance(x, float) and (x <=-999 or x == np.nan):
+                xstr='NaN'
+            elif isinstance(x, int):
+                xstr='%d'%x
+            else:
+                xstr='???'
+                
+            cur_tb.append(xstr)
+    elif nrows==1:
+        if isinstance(curexp, float):
+            cur_tb='%4.3f'%curexp
+            ref_tb='%2.1f'%ref
+        elif isinstance(curexp, int):
+            cur_tb='%d'%curexp
+            ref_tb=ref
+        else:
+            cur_tb=curexp
+            ref_tb=ref
+            
+
+    if qa in ['snr', 'integ']:
+        # per_TGT
+        key_tb= [qa_metrics[qa] + ' ( %s)'%objtype[i] for i in list(range(nrows))]
+       
+    elif qa in [ 'countpix','getbias','getrms']:
+        #per AMP
+        key_tb= [qa_metrics[qa] + ' (AMP %d)'%(i+1) for i in list(range(nrows))]
+        
+    elif qa in ['xwsigma']:
+        # per axis x or w
+        key_tb = [ qa_metrics[qa] + [' (x)',' (w)'][i] for i in list(range(nrows))]
+    else:
+        key_tb=key
+        #cur_tb=curexp
+        #ref_tb=ref
+                
+    
+    tblines=""
+    for i in  list(range(nrows)):
+        if nrows==1:
+            tblines=tblines+\
+                """<tr>
+                <td rowspan="{}">{}</td> <td>{}</td> <td>{}</td> <td>{}</td>
+                </tr>
+                """.format(nrows, comments, key, cur_tb, ref_tb)
+        elif i ==0:
+            tblines=tblines+\
+                """<tr>
+                <td rowspan="{}">{}</td> <td>{}</td> <td>{}</td> <td>{}</td>
+                </tr>
+                """.format(nrows, comments, key_tb[i], cur_tb[i], ref_tb[i])
+        else:
+            tblines=tblines+\
+                """<tr>
+                <td>{}</td> <td>{}</td> <td>{}</td>
+                </tr>
+                """.format( key_tb[i], cur_tb[i], ref_tb[i])
+
+    print(cur_tb, ref_tb)
+    return style + title + header + tblines + end
+
+
+
+
 def metric_table_OLD(metricname, comments, keyname, curexp='xxx', refexp='xxx'):
```

Comment:
This function seems to not be used anymore can you remove it?

Reply:
Sure. done!